finality-aleph/src/sync/forest/mod.rs

```diff
@@ -303,8 +291,10 @@ where
     }
 
     fn insert_id(&mut self, id: BlockIdFor<J>, holder: Option<I>) -> Result<(), Error> {
-        if let Some(state) = self.special_state(&id) {
-            return Err(Error::Special(state));
+        match self.special_state(&id) {
+            Some(SpecialState::TooNew) => return Err(Error::TooNew),
+            Some(_) => return Ok(()),
+            _ => {}
```

Comment:
Eh, at this point you can just put the code below here. Then you can delete the `return`s above, which is nice.

Reply:
ok, done